api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -77,51 +69,44 @@ public class AppFrameworkServiceImpl extends BaseOpenmrsService implements AppFr
 
 	private AppFrameworkConfig appFrameworkConfig;
 
-	private ScriptEngine javascriptEngine;
-
 	private AllUserApps allUserApps;
 
+	private org.graalvm.polyglot.Context context;
+
 	public AppFrameworkServiceImpl(AllAppTemplates allAppTemplates, AllAppDescriptors allAppDescriptors,
 	    AllFreeStandingExtensions allFreeStandingExtensions, AllComponentsState allComponentsState,
 	    LocationService locationService, FeatureToggleProperties featureToggles, AppFrameworkConfig appFrameworkConfig,
-	    AllUserApps allUserApps) throws ScriptException {
+	    AllUserApps allUserApps) {
 		this.allAppTemplates = allAppTemplates;
 		this.allAppDescriptors = allAppDescriptors;
 		this.allFreeStandingExtensions = allFreeStandingExtensions;
 		this.allComponentsState = allComponentsState;
 		this.locationService = locationService;
 		this.featureToggles = featureToggles;
 		this.appFrameworkConfig = appFrameworkConfig;
-		this.javascriptEngine = new ScriptEngineManager().getEngineByName("JavaScript");
 		this.allUserApps = allUserApps;
 
-		if (javascriptEngine == null) {
-			javascriptEngine = new jdk.nashorn.api.scripting.NashornScriptEngineFactory().getScriptEngine();
-			javascriptEngine.setBindings(javascriptEngine.createBindings(), ScriptContext.GLOBAL_SCOPE);
-		}
+		this.context = org.graalvm.polyglot.Context.newBuilder()
```

Comment:
What is this for?

Reply:
GraalVm uses polygot.Context to evaluate JS: https://docs.oracle.com/en/graalvm/enterprise/20/docs/reference-manual/embed-languages/